web/src/frontend/classic/stores/Helpers.ts

```diff
@@ -0,0 +1,47 @@
+import { writable } from 'svelte/store';
+import type { IValue, Setting} from '../../../engine/SettingsManager';
+import type { Key } from '../../../engine/SettingsGlobal';
+
+const globalsettings = HakuNeko.SettingsManager.OpenScope();
+
+/**
+ * Create a writable svelte store that is coupled with the given setting
+ * and updated whenever the value of the underlying setting is changed.
+ * @param setting - a specific setting of the frontend
+ */
+export function CreateWritableStore<T extends IValue>(setting:Setting<T>) {
+    const { subscribe, set } = writable(setting.Default);
+
+    setting.ValueChanged.Subscribe(
+        (_: typeof setting, value: T) => set(value)
+    );
+    return {
+        subscribe,
+        set: (value: T) => { setting.Value = value; set(value); },
+        reset: () => { setting.Value = setting.Default; set(setting.Default); },
+        setting : setting
+    };
+}
+
+/**
+ * Create a writable svelte store that is coupled with the given setting
+ * and updated whenever the value of the underlying setting is changed.
+ * @param settingKey - an existing key (created in the engine)
+ */
+export function GetEngineSetting<T extends IValue>(settingKey: Key) {
+    const setting: Setting<T> = globalsettings.Get(settingKey);
+    if (!setting) throw new Error(`Setting ${settingKey} does not exists`);
```

Comment:
Is this error redundant?
TypeScript is expected to fail compilation in case an invalid `Key` is used for `settingKey`.

Reply:
In case the key is correct but the setting couldn't be found in the setting manager (initialisation issue or any other reason) I wanted a way to make sure we see it.
So it was not intended as a compilation check but a runtime check.